src/array/empty.js

```diff
@@ -0,0 +1,23 @@
+define(['../lang/isArray'], function(isArray) {
+
+    /**
+     * Array empty
+     */
+
+    function empty(array) {
+        var i, len;
+
+        if(!isArray(array)) {
+            return [];
+        }
+
+        for (i = 0, len = array.length; i < len; i++) {
+            array.pop();
+        }
+
```

Comment:
Erm `array.length = 0` does empty the array and does allow for garbage collection .